### PR TITLE
KV Store: Initial graveler skeleton

### LINKED ISSUE #3568: KV Graveler Initial Implementation
Implement `graveler` `KV` support alongside `SQL`. `KV graveler` should not really use 'KV' at this point, but rather delegate all `DB` related calls to the `DB graveler`.
The purpose is to build a working skeleton as fast as possible, to allow breaking the `graveler KV` work itself into smaller pieces (which we can implement in parallel)
This task also include creating the followup implementation tasks

**DoD**
A working implementation of graveler over KV, toggled by KV feature flag, that delegates all calls to `DB graveler`
Followup tasks for the basic `KV` implementation

### PR DESCRIPTION
**Pull request by N-o-Z**
- Closes #3568

### CODE REVIEW

File: pkg/graveler/graveler_test.go [hunk 1]
```diff
@@ -136,34 +139,40 @@ func (h *Hooks) NewRunID() string {
...
 }
 
-func TestGraveler_List(t *testing.T) {
+type gravelerFunc func(committedManager graveler.CommittedManager,
+	stagingManager graveler.StagingManager,
+	refManager graveler.RefManager,
+	gcManager graveler.GarbageCollectionManager,
+	protectedBranchesManager graveler.ProtectedBranchesManager) catalog.Store
+
+func TestDBGraveler_List(t *testing.T) {
```
Comment (itaiad200) [inline pkg/graveler/graveler_test.go:148]: Please reuse the same code tests and avoid copying it.

**Review by itaiad200 — CHANGES_REQUESTED** [review]
LGTM - Just reuse the tests.

**Review by eden-ohana — APPROVED** [review]
LGTM

**Review by itaiad200 — APPROVED** [review]

**Comment by N-o-Z** [comment]
@itaiad200  Thanks.
Please note that it seems the issue with the migration test is not resolved:
https://github.com/treeverse/lakeFS/runs/7169497154?check_suite_focus=true

```
lakefs_1    | time="2022-07-03T14:46:01Z" level=error msg="error making request" func="pkg/block/s3.(*Adapter).streamToS3" file="build/pkg/block/s3/adapter.go:232" error="Put \"[https://esti-system-testing.s3.amazonaws.com/1220/migrate/6058/cb0qn9h2igr59u77c58g/testimport/_lakefs/3bcd5ee790dbc4318c09ae4aded17db86c92d3179c45ed70b31be3822bff897d\](https://esti-system-testing.s3.amazonaws.com/1220/migrate/6058/cb0qn9h2igr59u77c58g/testimport/_lakefs/3bcd5ee790dbc4318c09ae4aded17db86c92d3179c45ed70b31be3822bff897d/)": net/http: HTTP/1.x transport connection broken: read tcp 172.18.0.3:48412->52.216.234.19:443: read: connection reset by peer" host="lakefs:8000" log_audit=API method=POST operation=PutObject path=/api/v1/repositories/testimport/branches/ranges request_id=f4aef10e-25e0-4010-a426-1c615b032f44 service_name=rest_api url="https://esti-system-testing.s3.amazonaws.com/1220/migrate/6058/cb0qn9h2igr59u77c58g/testimport/_lakefs/3bcd5ee790dbc4318c09ae4aded17db86c92d3179c45ed70b31be3822bff897d"
```

**Pull request merged**